### PR TITLE
Make rootgrid an unique_ptr

### PR DESCRIPTION
**Pull request by tobiolo**
This is the second approach. I used the wrong comment type (review comment), so maybe this more cleaner now.

### CODE REVIEW

File: src/document.h [hunk 1]
```diff
@@ -570,7 +568,7 @@ struct Document {
         ResetFont();
         dc.SetUserScale(1, 1);
         curdrawroot = WalkPath(drawpath);
-        int psb = curdrawroot == rootgrid ? 0 : curdrawroot->MinRelsize();
+        int psb = curdrawroot == rootgrid.get() ? 0 : curdrawroot->MinRelsize();
```
Comment (tobiolo) [inline src/document.h:571]: We need to call `.get()` because curdrawroot is `Cell *` and rootgrid is `unique_ptr<Cell>`.

Comment (aardappel) [reply src/document.h:571]: https://en.cppreference.com/w/cpp/memory/unique_ptr/operator_bool

File: src/system.h [hunk 2]
```diff
@@ -464,13 +464,12 @@ struct System {
...
                     if (!doc.Load(fn)) goto problem;
-                    Cell *&r = InitDB(1);
+                    unique_ptr<Cell> &r = InitDB(1);
                     Cell *c = *r->grid->cells;
                     FillXML(c, doc.GetRoot(), k == A_IMPXMLA);
                     if (!c->HasText() && c->grid) {
                         *r->grid->cells = nullptr;
-                        delete r;
-                        r = c;
+                        r.reset(c);
```
[the remark this answers is not in the document]
Comment (aardappel) [reply src/system.h:472]: this seems fine

File: src/system.h [hunk 3]
```diff
@@ -487,7 +486,7 @@ struct System {
 
                     if (as.size()) switch (k) {
                             case A_IMPTXTI: {
-                                Cell *r = InitDB(1);
+                                Cell *r = InitDB(1).get();
```
[the remark this answers is not in the document]
Comment (aardappel) [reply src/system.h:489]: this IS borrowing a pointer

File: src/grid.h [hunk 4]
```diff
@@ -608,7 +608,7 @@ struct Grid {
         const int root_grid_spacing = 2;  // Can't be adjusted in editor, so use a default.
         const int font_size = 14 - indent / 2;
         const int grid_border_width =
-            cell == doc->rootgrid ? root_grid_spacing : user_grid_outer_spacing - 1;
+            cell == doc->rootgrid.get() ? root_grid_spacing : user_grid_outer_spacing - 1;
```
Comment (aardappel) [inline src/grid.h:611]: See my other comment about using it as boolean

File: src/document.h (hunk 1 above)
Comment (tobiolo) [reply src/document.h:571]: I think the precedence order here is:
`int psb = ((curdrawroot == rootgrid.get()) ? 0 : curdrawroot->MinRelsize());`
Is this correct? If yes, I cannot compare `curdrawroot` and `rootgrid` directly?

File: src/grid.h (hunk 4 above)
Comment (tobiolo) [reply src/grid.h:611]: Also here:
I think the precedence order is 
`const int grid_border_width = ((cell == doc->rootgrid.get()) ? root_grid_spacing : user_grid_outer_spacing) - 1)`

File: src/document.h (hunk 1 above)
Comment (aardappel) [reply src/document.h:571]: Ah, did not see. You're right!

File: src/grid.h (hunk 4 above)
Comment (tobiolo) [reply src/grid.h:611]: Unfortunately they have not, would make things easier...

**Comment by tobiolo** [comment]
Thanks for the merge!

**Pull request merged**